### ISSUE TITLE
Suggestion: new rule action for 'setting "their" account to...'

### ISSUE THREAD

**Issue by SplinterHead** [firefly-iii/firefly-iii#7462]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/more-information/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description

I've just got set up with Firefly and I'm currently creating rules to tidy up my transaction history where I have a lot of `(unknown source account)` and `(unknown destination account)`, which means I am creating two rules for each account to deal with withdrawals and deposits. 

I may have missed a unified rule that would allow setting the Source account on Deposit, and also setting the Destination account on Withdrawal

### Solution

A single rule action called something like `Set external account to...` which will set the Source account on a Deposit and set the Destination account on Withdrawal.

I read the "Very good idea, unfortunately" and wasn't sure if this fell in the if-else realm for the rule engine

### What are alternatives?

Currently create two individual rules

For Deposits use the `Set Source account to...` action
For Withdrawals use the `Set Destination account to...` action

**Comment by JC5** [comment]
Trouble is, it's hard to decide which one it's going to be in case of transfers.

**Comment by SplinterHead** [comment]
True, transfers would be an issue. I guess it could have validation that the action only be used with the `Transaction is of type...` trigger, and then only when that is set to Withdrawal or Deposit. 

I see the complexity growing 😬

**Comment by JC5** [comment]
Yep, that's my challenge here 😬 . I want people to get it without having to use the manual. Your solution (two rules) is something most people can organically come up with. Almost all rule actions and triggers are so "easy" that it's pretty doable to set something up without reading more about it.
